### ISSUE TITLE
The + character breaks search

### ISSUE THREAD

**Issue by sarahfg** [Automattic/simplenote-electron#1945]
## Expected
When searching for the character `+` in a note all notes with that character should be listed

### Observed
When you type `+` into the search bar the app goes blank. It must be exited and restarted to be useable

### Reproduced
Reported on Windows using 1.14
Reproduced on Mac OS Catalina 10.15.3 on 1.14ION

**Comment by dmsnell** [comment]
Just tested this on https://app.simplenote.com on my account and I found the expected notes with a `+` in them, highlighted as expected. Also I tested in `develop` and things worked as expected.

@sarahfg can we see if there are error messages in the developer console? Please reach out to me if you want some help.

**Comment by sarahfg** [comment]
Perhaps the difference is that I'm testing on Electron running on a Mac. Simplenote Version 1.14 Mac OS 10.15.3 . It works as expected for me in a web browser.

Here's what I see in the developer console on my Mac
```
app.ba12d056fe9f225dbbbd.js:99 SyntaxError: Invalid regular expression: /(tag:)?+/: Nothing to repeat
    at new RegExp (<anonymous>)
    at note-list.125d91ad80b1e204fd83.js:1
    at Array.filter (<anonymous>)
    at note-list.125d91ad80b1e204fd83.js:1
    at K (note-list.125d91ad80b1e204fd83.js:1)
    at Function.t.default.n [as mapToProps] (note-list.125d91ad80b1e204fd83.js:1)
    at r (app.ba12d056fe9f225dbbbd.js:14)
    at h (app.ba12d056fe9f225dbbbd.js:14)
    at app.ba12d056fe9f225dbbbd.js:14
    at app.ba12d056fe9f225dbbbd.js:14
du @ app.ba12d056fe9f225dbbbd.js:99
n.callback @ app.ba12d056fe9f225dbbbd.js:99
_i @ app.ba12d056fe9f225dbbbd.js:99
Ei @ app.ba12d056fe9f225dbbbd.js:99
Es @ app.ba12d056fe9f225dbbbd.js:99
t.unstable_runWithPriority @ app.ba12d056fe9f225dbbbd.js:107
Yo @ app.ba12d056fe9f225dbbbd.js:99
ws @ app.ba12d056fe9f225dbbbd.js:99
us @ app.ba12d056fe9f225dbbbd.js:99
(anonymous) @ app.ba12d056fe9f225dbbbd.js:99
t.unstable_runWithPriority @ app.ba12d056fe9f225dbbbd.js:107
Yo @ app.ba12d056fe9f225dbbbd.js:99
Xo @ app.ba12d056fe9f225dbbbd.js:99
Qo @ app.ba12d056fe9f225dbbbd.js:99
(anonymous) @ app.ba12d056fe9f225dbbbd.js:99
ce @ app.ba12d056fe9f225dbbbd.js:99
de @ app.ba12d056fe9f225dbbbd.js:99
Nn @ app.ba12d056fe9f225dbbbd.js:99
An @ app.ba12d056fe9f225dbbbd.js:99
Rn @ app.ba12d056fe9f225dbbbd.js:99
t.unstable_runWithPriority @ app.ba12d056fe9f225dbbbd.js:107
Yo @ app.ba12d056fe9f225dbbbd.js:99
se @ app.ba12d056fe9f225dbbbd.js:99
jn @ app.ba12d056fe9f225dbbbd.js:99
app.ba12d056fe9f225dbbbd.js:99 Uncaught SyntaxError: Invalid regular expression: /(tag:)?+/: Nothing to repeat
    at new RegExp (<anonymous>)
    at note-list.125d91ad80b1e204fd83.js:1
    at Array.filter (<anonymous>)
    at note-list.125d91ad80b1e204fd83.js:1
    at K (note-list.125d91ad80b1e204fd83.js:1)
    at Function.t.default.n [as mapToProps] (note-list.125d91ad80b1e204fd83.js:1)
    at r (app.ba12d056fe9f225dbbbd.js:14)
    at h (app.ba12d056fe9f225dbbbd.js:14)
    at app.ba12d056fe9f225dbbbd.js:14
    at app.ba12d056fe9f225dbbbd.js:14
```

![](https://d.pr/i/wgxoFb+)

(Image link: https://d.pr/i/wgxoFb )

**Comment by sarahfg** [comment]
There's another report of a similar issue with the `\` character in electron 1.14 in Linux breaking search in the same way. I can reproduce this issue with Electron 1.14 in MacOS.

Here is the debug log: 

![](https://d.pr/i/hzgFXu+)

(Image link: https://d.pr/i/hzgFXu )

```
Simperium config: Production
app.ba12d056fe9f225dbbbd.js:99 SyntaxError: Invalid regular expression: /(tag:)?\/: \ at end of pattern
    at new RegExp (<anonymous>)
    at note-list.125d91ad80b1e204fd83.js:1
    at Array.filter (<anonymous>)
    at note-list.125d91ad80b1e204fd83.js:1
    at K (note-list.125d91ad80b1e204fd83.js:1)
    at Function.t.default.n [as mapToProps] (note-list.125d91ad80b1e204fd83.js:1)
    at r (app.ba12d056fe9f225dbbbd.js:14)
    at h (app.ba12d056fe9f225dbbbd.js:14)
    at app.ba12d056fe9f225dbbbd.js:14
    at app.ba12d056fe9f225dbbbd.js:14
du @ app.ba12d056fe9f225dbbbd.js:99
n.callback @ app.ba12d056fe9f225dbbbd.js:99
_i @ app.ba12d056fe9f225dbbbd.js:99
Ei @ app.ba12d056fe9f225dbbbd.js:99
Es @ app.ba12d056fe9f225dbbbd.js:99
t.unstable_runWithPriority @ app.ba12d056fe9f225dbbbd.js:107
Yo @ app.ba12d056fe9f225dbbbd.js:99
ws @ app.ba12d056fe9f225dbbbd.js:99
us @ app.ba12d056fe9f225dbbbd.js:99
(anonymous) @ app.ba12d056fe9f225dbbbd.js:99
t.unstable_runWithPriority @ app.ba12d056fe9f225dbbbd.js:107
Yo @ app.ba12d056fe9f225dbbbd.js:99
Xo @ app.ba12d056fe9f225dbbbd.js:99
Qo @ app.ba12d056fe9f225dbbbd.js:99
(anonymous) @ app.ba12d056fe9f225dbbbd.js:99
ce @ app.ba12d056fe9f225dbbbd.js:99
de @ app.ba12d056fe9f225dbbbd.js:99
Nn @ app.ba12d056fe9f225dbbbd.js:99
An @ app.ba12d056fe9f225dbbbd.js:99
Rn @ app.ba12d056fe9f225dbbbd.js:99
t.unstable_runWithPriority @ app.ba12d056fe9f225dbbbd.js:107
Yo @ app.ba12d056fe9f225dbbbd.js:99
se @ app.ba12d056fe9f225dbbbd.js:99
jn @ app.ba12d056fe9f225dbbbd.js:99
app.ba12d056fe9f225dbbbd.js:99 Uncaught SyntaxError: Invalid regular expression: /(tag:)?\/: \ at end of pattern
    at new RegExp (<anonymous>)
    at note-list.125d91ad80b1e204fd83.js:1
    at Array.filter (<anonymous>)
    at note-list.125d91ad80b1e204fd83.js:1
    at K (note-list.125d91ad80b1e204fd83.js:1)
    at Function.t.default.n [as mapToProps] (note-list.125d91ad80b1e204fd83.js:1)
    at r (app.ba12d056fe9f225dbbbd.js:14)
    at h (app.ba12d056fe9f225dbbbd.js:14)
    at app.ba12d056fe9f225dbbbd.js:14
    at app.ba12d056fe9f225dbbbd.js:14
(anonymous) @ note-list.125d91ad80b1e204fd83.js:1
(anonymous) @ note-list.125d91ad80b1e204fd83.js:1
K @ note-list.125d91ad80b1e204fd83.js:1
t.default.n @ note-list.125d91ad80b1e204fd83.js:1
r @ app.ba12d056fe9f225dbbbd.js:14
h @ app.ba12d056fe9f225dbbbd.js:14
(anonymous) @ app.ba12d056fe9f225dbbbd.js:14
(anonymous) @ app.ba12d056fe9f225dbbbd.js:14
useMemo @ app.ba12d056fe9f225dbbbd.js:99
useMemo @ app.ba12d056fe9f225dbbbd.js:91
d @ app.ba12d056fe9f225dbbbd.js:14
pa @ app.ba12d056fe9f225dbbbd.js:99
Ga @ app.ba12d056fe9f225dbbbd.js:99
Ya @ app.ba12d056fe9f225dbbbd.js:99
Cu @ app.ba12d056fe9f225dbbbd.js:99
ys @ app.ba12d056fe9f225dbbbd.js:99
vs @ app.ba12d056fe9f225dbbbd.js:99
us @ app.ba12d056fe9f225dbbbd.js:99
(anonymous) @ app.ba12d056fe9f225dbbbd.js:99
t.unstable_runWithPriority @ app.ba12d056fe9f225dbbbd.js:107
Yo @ app.ba12d056fe9f225dbbbd.js:99
Xo @ app.ba12d056fe9f225dbbbd.js:99
Qo @ app.ba12d056fe9f225dbbbd.js:99
(anonymous) @ app.ba12d056fe9f225dbbbd.js:99
ce @ app.ba12d056fe9f225dbbbd.js:99
de @ app.ba12d056fe9f225dbbbd.js:99
Nn @ app.ba12d056fe9f225dbbbd.js:99
An @ app.ba12d056fe9f225dbbbd.js:99
Rn @ app.ba12d056fe9f225dbbbd.js:99
t.unstable_runWithPriority @ app.ba12d056fe9f225dbbbd.js:107
Yo @ app.ba12d056fe9f225dbbbd.js:99
se @ app.ba12d056fe9f225dbbbd.js:99
jn @ app.ba12d056fe9f225dbbbd.js:99
```

**Comment by codebykat** [comment]
I've just tried to reproduce this in various places and cannot; though I don't have a Windows instance to test on, I can successfully search for both `+` and `\` characters in Electron and web on Mac OS. It also works for tags that contain these special characters. I think it was probably fixed by #1826 or a related fix; if we get more reports on the current version let's reopen.
